Fixed two more occurrences of `requires public`

jklingen opened the pull request:
Changed to `requires transitive`

jklingen committed:
Fixed two more occurrences of `requires public`

Changed to `requires transitive`

nipafx commented on README.md:
```diff
@@ -54,7 +54,7 @@ Nothing changed [from before](https://github.com/CodeFX-org/demo-jigsaw-advent-c
 ### _chocolate_
 
-As [before with _factories_](https://github.com/CodeFX-org/demo-jigsaw-advent-calendar/tree/02-splitting-into-modules#factories) this module must `require public` the _surprise_ module.
+As [before with _factories_](https://github.com/CodeFX-org/demo-jigsaw-advent-calendar/tree/02-splitting-into-modules#factories) this module must `require transitive` the _surprise_ module.
```
Could you also fix "require" to "requires"?

jklingen replied:
I assumed the "require" to be intentional, for the sake of grammatical correctness ("the module must require..."). But now that I think about it I agree that this might be confusing... maybe it would be better not to include "require" in backticks?

nipafx replied:
Since this is not the topic this branch is teaching, we can just drop the keywords and go with "...this module must require the _surprise_ module transitively".

nicolaiparlog commented:
@jklingen Thank you very much for your PRs. I didn't know anybody was actually going through this repo - awesome! :smiley:

jklingen commented:
@nicolaiparlog thanks to you for making this 😃. I thought it was good to do some warm-up exercises before testing migration of a real-world project and found this repo a really good starting point. 👍

jklingen committed:
Remove code markdown and "transitive" from text

To avoid confusion due to English grammar of the sentence not matching `requires transitive` syntax.

jklingen commented:
Good idea, I just changed it as suggested.
